### ISSUE TITLE
Provide a DNS cluster peer discovery backend in the core

### ISSUE THREAD

**Issue by michaelklishin** [rabbitmq/rabbitmq-server#988]
Follow-up to #486. Since the DNS backend in rabbitmq-autocluster doesn't have any dependencies outside of Erlang/OTP, we can move it into the core.

This is a separate issue to keep #486 as focussed as possible. For example, we'll have to come up with a config syntax for this after #987 is resolved. This should not hold most of the work in #486 back.
